table/tables/tables_test.go

```diff
@@ -616,3 +622,50 @@ func (ts *testSuite) TestAddRecordWithCtx(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(txn.Commit(context.Background()), IsNil)
 }
+
+func (ts *testSuite) TestConstraintCheckForUniqueIndex(c *C) {
+	tk := testkit.NewTestKit(c, ts.store)
+	tk.MustExec("set @@tidb_constraint_check_in_place = 0")
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(id int(11) NOT NULL AUTO_INCREMENT,k int(11) NOT NULL DEFAULT '0',c char(120) NOT NULL DEFAULT '',PRIMARY KEY (id),UNIQUE KEY k_1 (k,c))")
+	tk.MustExec("insert into t(k,c) values(1, 'tidb')")
+	tk.MustExec("insert into t(k,c) values(2, 'tidb')")
+	tk.MustExec("begin")
+	tk.MustExec("update t set k=1 where id=2")
+	_, err := tk.Exec("commit")
+	c.Assert(err.Error(), Equals, "previous statement: update t set k=1 where id=2: [kv:1062]Duplicate entry '1-tidb' for key 'k_1'")
+	tk.MustExec("rollback")
+
+	tk.MustExec("set @@tidb_constraint_check_in_place = 1")
+	tk.MustExec("begin")
+	_, err = tk.Exec("update t set k=1 where id=2")
+	c.Assert(err.Error(), Equals, "[kv:1062]Duplicate entry '1-tidb' for key 'k_1'")
+	tk.MustExec("rollback")
+
+	tk1 := testkit.NewTestKit(c, ts.store)
+	tk2 := testkit.NewTestKit(c, ts.store)
+	tk1.MustExec("set @@tidb_txn_mode = 'pessimistic'")
+	tk1.MustExec("set @@tidb_constraint_check_in_place = 0")
+	tk2.MustExec("set @@tidb_txn_mode = 'pessimistic'")
+	tk1.MustExec("use test")
+	tk1.MustExec("begin")
+	tk1.MustExec("update t set k=3 where id=2")
+
+	ch := make(chan int, 2)
+	go func() {
+		tk2.MustExec("use test")
+		tk2.Exec("insert into t(k,c) values(3, 'tidb')")
+		ch <- 2
+	}()
+	time.Sleep(100000000)
```

Comment:
`100*time.Millisecond` is much more readable than 100000000

Reply:
done